VideoWeb/VideoWeb/ClientApp/src/app/shared/select-media-devices/select-media-devices.component.ts

```diff
@@ -99,18 +101,23 @@ export class SelectMediaDevicesComponent implements OnInit, OnDestroy {
         return this.selectedMediaDevicesForm.get('microphone');
     }
 
-    onSubmit() {
-        if (this.selectedMediaDevicesForm.invalid) {
-            return;
-        }
+    onChangeDevice() {
+        this.saveSelectedDevices();
+    }
+
+    private saveSelectedDevices() {
+        // save on select device
         const selectedCam = this.getSelectedCamera();
         const selectedMic = this.getSelectedMicrophone();
-        this.userMediaStreamService.stopStream(this.preferredCameraStream);
```

Comment:
have you assessed the impact of not stopping a stream when a user does the self test?